AEPAssurance/Source/AssuranceSessionOrchestrator.swift

```diff
@@ -24,68 +40,114 @@ class AssuranceSessionOrchestrator: AssurancePresentationDelegate {
 
     init(stateManager: AssuranceStateManager) {
         self.stateManager = stateManager
+        outboundEventBuffer = ThreadSafeArray(identifier: "Session Orchestrator's OutboundBuffer array")
     }
 
-    func createSession() {
+    /// Creates and starts a new  `AssuranceSession` with the provided `AssuranceSessionDetails`.
+    ///
+    /// A new AssuranceSession is only created if there isn't an already existing session.
+    /// Calling this method also shares the shared state for the extension with the provided session details
+    ///
+    /// - Parameters:
+    ///    - sessionDetails: An `AssuranceSessionDetails` instance containing all the essential data for starting a session
+    func createSession(withDetails sessionDetails: AssuranceSessionDetails) {
+        if let _ = session {
+            Log.warning(label: AssuranceConstants.LOG_TAG, "An active session already exists. Cannot create a new one.")
+            return
+        }
+
+        stateManager.shareAssuranceState(withSessionID: sessionDetails.sessionId)
+        session = AssuranceSession(sessionDetails: sessionDetails, stateManager: stateManager, sessionOrchestrator: self, outboundEvents: outboundEventBuffer)
+        session?.startSession()
 
+        outboundEventBuffer?.clear()
+        outboundEventBuffer = nil
     }
 
+    ///
+    /// Dissolve the active session (if one exists) and its associated states.
+    ///
     func terminateSession() {
+        hasEverTerminated = true
+        self.outboundEventBuffer = nil
+
+        stateManager.clearAssuranceState()
 
+        session?.disconnect()
+        session = nil
     }
 
-    func shutDownSession() {
+    func queueEvent(_ assuranceEvent: AssuranceEvent) {
+        /// Queue this event to the active session if one exists.
+        if let session = session {
+            session.sendEvent(assuranceEvent)
+            return
+        }
 
-    }
+        /// drop the event if outboundEventBuffer is nil
+        guard let outBoundEventBuffer = outboundEventBuffer else {
```

Comment:
nit: prefer if let here with the append in the block, one less return makes this code easier to read.